worksheet.rb

```diff
@@ -23,12 +37,124 @@
 # into this data structure, such as "DR0004"
 # and "3rd Feb 2016" and "RD0022"
 
+# data structure blueprint
+# [
+#   {
+#       driver_id: str,
+#       date: [int, int, int],
+#       cost: int,
+#       rider_id: str,
+#       rating: int
+#   }
+# ]
+
+rides_data = [
+    ['DRIVER_ID','DATE','COST','RIDER_ID','RATING'],
+    ['DR0004','3rd Feb 2016','5','RD0022','5'],
+    ['DR0001','3rd Feb 2016','10','RD0003','3'],
+    ['DR0002','3rd Feb 2016','25','RD0073','5'],
+    ['DR0001','3rd Feb 2016','30','RD0015','4'],
+    ['DR0003','4th Feb 2016','5','RD0066','5'],
+    ['DR0004','4th Feb 2016','10','RD0022','4'],
+    ['DR0002','4th Feb 2016','15','RD0013','1'],
+    ['DR0003','5th Feb 2016','50','RD0003','2'],
+    ['DR0002','5th Feb 2016','35','RD0066','3'],
+    ['DR0004','5th Feb 2016','20','RD0073','5'],
+    ['DR0001','5th Feb 2016','45','RD0003','2']
+]
+
+# return an array with integer representation of dates
+def line_break
+  puts '--------------------------------------------------------'
+end
+
+def parse_date(date_string)
+  day_month_year = []
+  d = Date.parse(date_string)
+  return day_month_year << d.mday  <<  d.mon  << d.year
+end
+
+# create the data structure from the blueprint
+def structure_ride_share(data)
+  top_array = []
+  # create default hash keys based on column headings
+  headings = data[0].map { |heading| heading.downcase.to_sym }
+
+  (data.length - 1).times do |index|
+    #skip heading
+    index += 1
+
+    #choose row of data
+    row = data[index]
+
+    #populate ride hashes - really long!
+    ride = Hash[headings[0], row[0], headings[1], parse_date(row[1]), headings[2], row[2].to_i, headings[3], row[3], headings[4], row[4].to_i]
+    top_array << ride
+  end
+  return top_array
+end
+
+ride_share_data = structure_ride_share(rides_data)
+line_break
+puts 'Ride Share Data:'
+line_break
+pp ride_share_data
```

Comment:
Just for readability it's best to put your methods at the top of the file and the main program code all together at the bottom.  

In reading this I was seeing

[method definitions]
some lines in the program here
[more method definitions]
more lines in the main program

So the methods here break up the flow of the regular program and make it harder to trace.

Reply:
Thanks for the feedback on this! I will re-arrange!